directembedding/src/main/scala/ch/epfl/directembedding/Utils.scala

```diff
@@ -10,11 +12,22 @@ trait MacroModule {
 }
 
 trait DirectEmbeddingModule extends MacroModule {
-  import c.universe._
-  def lift(tree: Tree): Tree
+  val dslName: String
+  val dslEndpointMethod: String
+  val virtualizationConfig: String
+  val liftMethodName: String
+  val logLevel: Int
 }
 
-trait DirectEmbeddingUtils {
-  def debugLevel: Int
-  def log(s: => String, level: Int = 0) = if (debugLevel < level) println(s)
+trait DirectEmbeddingUtils extends MacroModule with TransformationUtils {
+  import c.universe._
+
+  def logTree(t: Tree, level: Int = 0) = {
+    log(s"$t", level)
+    log(showRaw(t, printTypes = true), level + 1)
+  }
+  def logIndented[T](e: T, indent: Int, level: Int = 0) = if (debugLevel > level) {
```

Comment:
This is better. The problem was that `s"e"` does not print the message, if I am not mistaken.